controllers/cluster_controller_phases.go

```diff
@@ -174,12 +174,12 @@ func (r *ClusterReconciler) reconcileInfrastructure(ctx context.Context, cluster
 	}
 
 	// Get and parse Status.APIEndpoint field from the infrastructure provider.
-	if len(cluster.Status.APIEndpoints) == 0 {
-		if err := util.UnstructuredUnmarshalField(infraConfig, &cluster.Status.APIEndpoints, "status", "apiEndpoints"); err != nil {
-			return errors.Wrapf(err, "failed to retrieve Status.APIEndpoints from infrastructure provider for Cluster %q in namespace %q",
+	if cluster.Spec.ControlPlaneEndpoint.IsZero() {
+		if err := util.UnstructuredUnmarshalField(infraConfig, &cluster.Spec.ControlPlaneEndpoint, "spec", "controlPlaneEndpoint"); err != nil {
+			return errors.Wrapf(err, "failed to retrieve Spec.ControlPlaneEndpoint from infrastructure provider for Cluster %q in namespace %q",
 				cluster.Name, cluster.Namespace)
-		} else if len(cluster.Status.APIEndpoints) == 0 {
-			return errors.Wrapf(err, "retrieved empty Status.APIEndpoints from infrastructure provider for Cluster %q in namespace %q",
+		} else if cluster.Spec.ControlPlaneEndpoint.IsZero() {
+			return errors.Wrapf(err, "retrieved empty Spec.ControlPlaneEndpoint from infrastructure provider for Cluster %q in namespace %q",
```

Comment:
Do we really want to return an error if the referenced infra spec.controlPlaneEndpoint is unset? Could we return `nil` instead?

Reply:
Removed, this only made sense before given that the field was a list